Remove collection share access column header icon

Shrinks99 opened the pull request:
Closes #1351

### Changes
- Removes the collection share access icon's header icon which was just an icon.  This is now mostly in line with how we display status icons in the archived items list (there is a spacing difference between the two lists regarding the placement of the icon vs the label and its alignment with the text (here) vs the icon (archived items list).

### Screenshots

Before / After

<img width="391" alt="Screenshot 2023-11-13 at 4 41 45 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/5672810/51093f1d-497f-430f-a8b3-fe9e82356b3f">
<img width="356" alt="Screenshot 2023-11-13 at 4 41 31 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/5672810/16f8a290-1f09-4701-84af-63fb4fd6f8c9">

Shrinks99 committed:
Remove collection share access column header icon

emma-sg commented on frontend/src/pages/org/collections-list.ts:
```diff
@@ -382,13 +382,7 @@ export class CollectionsList extends LiteElement {
-              ></sl-icon>
-            </div>
+            <div class="col-span-1"></div>
             <div class="col-span-1 text-xs">${msg("Name")}</div>
```
```suggestion
            <div class="col-span-1 row-start-2 text-xs">${msg("Name")}</div>
```
I think this should work — maybe double check though? But I don't think we need the empty div

Shrinks99 replied:
Not quite but you're right, empty div is a very dumb way of doing this.

\>_<

Fixed! 🙃

Shrinks99 committed:
Remove useless grid item

You ever have a day so long you forget how to actually use CSS grid?

Thanks @emma-sg
